### PR TITLE
That time DevUtils wouldn't die

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Description

Adds the following `DevUtils`-equivalent functions to `NativeOrdersFeature` in the EP:
- `getLimitOrderRelevantState()`
- `getRfqOrderRelevantState()`
- `batchGetLimitOrderRelevantStates()`
- `batchGetRfqOrderRelevantStates()`

Also updated the return data size checks in `LibERC20TokenV06` to allow for tokens that return >= 32 bytes.

#### TODO
- [x] tests
- [x] Regenerate contract-wrappers

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/zero-ex/contracts/src/features/INativeOrdersFeature.sol [hunk 1]
```diff
@@ -346,4 +346,81 @@ interface INativeOrdersFeature {
...
+    function batchGetLimitOrderRelevantState(
+        LibNativeOrder.LimitOrder[] calldata orders,
+        LibSignature.Signature[] calldata signatures
+    )
+        external
+        view
+        returns (
+            LibNativeOrder.OrderInfo[] memory orderInfos,
+            uint128[] memory actualFillableTakerTokenAmounts,
+            bool[] memory isSignatureValids
```
Comment (dorothy-zbornak) [inline contracts/zero-ex/contracts/src/features/INativeOrdersFeature.sol:405]: It's called english, look it up.

File: contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol [hunk 2]
```diff
@@ -751,6 +907,48 @@ contract NativeOrdersFeature is
...
+            return 0;
+        }
+        if (params.orderInfo.takerTokenFilledAmount >= params.orderTakerAmount) {
+            // Already fully filled.
+            return 0;
+        }
+
+        // Get the fillable maker amount based on the order quantities and
+        // previously filled amount
+        uint256 fillableMakerTokenAmount = LibMathV06.getPartialAmountFloor(
```
Comment (hysz) [inline contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol:930]: Rounding looks good +1

File: contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol [hunk 3]
```diff
@@ -751,6 +907,48 @@ contract NativeOrdersFeature is
...
+            uint256(params.orderMakerAmount)
+        );
+        // Clamp it to the amount of maker tokens we can spend on behalf of the
+        // maker.
+        fillableMakerTokenAmount = LibSafeMathV06.min256(
+            fillableMakerTokenAmount,
+            _getSpendableERC20BalanceOf(params.makerToken, params.maker)
+        );
+        // Conver to taker token amount.
+        return uint128(LibMathV06.getPartialAmountCeil(
```
Comment (hysz) [inline contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol:945]: Is it possible this overflows; should we do a safe cast?

File: contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol [hunk 4]
```diff
@@ -751,6 +907,48 @@ contract NativeOrdersFeature is
...
+        private
+        view
+        returns (uint128 actualFillableTakerTokenAmount)
+    {
+        if (params.orderMakerAmount == 0 || params.orderTakerAmount == 0) {
+            // Empty order.
+            return 0;
+        }
+        if (params.orderInfo.takerTokenFilledAmount >= params.orderTakerAmount) {
+            // Already fully filled.
```
Comment (hysz) [inline contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol:924]: Should we check if the order status is `FILLABLE`? Could be cancelled or just invalid.

**Review by hysz — COMMENTED** [review]
A few minor questions. Looks good to merge after!

File: contracts/zero-ex/contracts/src/features/INativeOrdersFeature.sol (hunk 1 above)
Comment (mzhu25) [reply contracts/zero-ex/contracts/src/features/INativeOrdersFeature.sol:405]: `areSignaturesValid`? 🤷

File: contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol (hunk 4 above)
Comment (dorothy-zbornak) [reply contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol:924]: Oh, doh, good catch.

File: contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol [hunk 5]
```diff
@@ -751,6 +907,48 @@ contract NativeOrdersFeature is
...
+            uint256(params.orderTakerAmount),
+            uint256(params.orderMakerAmount)
+        );
+        // Clamp it to the amount of maker tokens we can spend on behalf of the
+        // maker.
+        fillableMakerTokenAmount = LibSafeMathV06.min256(
+            fillableMakerTokenAmount,
+            _getSpendableERC20BalanceOf(params.makerToken, params.maker)
+        );
+        // Conver to taker token amount.
```
Comment (mzhu25) [inline contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol]:
```suggestion
        // Convert to taker token amount.
```

**Review by mzhu25 — APPROVED** [review]
lgtm, nice tests

**Review by hysz — APPROVED** [review]
Looks great, let's merge!

**Pull request merged**